internal/providers/terraform/azure/cdn_endpoint.go

```diff
@@ -0,0 +1,297 @@
+		}
+		if d.Get("delivery_rule").Type != gjson.Null {
+			numberOfRules += len(d.Get("delivery_rule").Array())
+		}
+
+		if numberOfRules > 5 {
+			numberOfRules -= 5
+
+			costComponents = append(costComponents, cdnCostComponent(
+				"Rules engine rule (over 5)",
+				"rules",
+				location,
+				"Azure CDN from Microsoft",
+				"Standard",
+				"Rules",
+				"5",
+				decimalPtr(decimal.NewFromInt(int64(numberOfRules))),
+			))
+		}
+
+		if numberOfRules > 0 {
+			var rulesRequests *decimal.Decimal
+			if u != nil && u.Get("monthly_rules_engine_requests").Type != gjson.Null {
+				rulesRequests = decimalPtr(decimal.NewFromInt(u.Get("monthly_rules_engine_requests").Int() / 1000000))
+			}
+			costComponents = append(costComponents, cdnCostComponent(
+				"Rules engine requests",
+				"1M requests",
+				location,
+				"Azure CDN from Microsoft",
+				"Standard",
+				"Requests",
+				"0",
+				rulesRequests,
+			))
+		}
+	}
+
+	if strings.ToLower(sku) == "standard_akamai" || strings.ToLower(sku) == "standard_verizon" {
+		if d.Get("optimization_type").Type != gjson.Null {
+			if strings.ToLower(d.Get("optimization_type").String()) == "dynamicsiteacceleration" {
+				costComponents = append(costComponents, cdnAccelerationDataTransfersCostComponents(location, sku, d, u)...)
+			}
+		}
+	}
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+	}
+}
+
+func cdnOutboundDataCostComponents(location, sku string, u *schema.UsageData) []*schema.CostComponent {
+	costComponents := []*schema.CostComponent{}
+
+	type dataTier struct {
+		name       string
+		startUsage string
+	}
+
+	var name, productName, skuName, meterName string
+	if s := strings.Split(sku, "_"); len(s) == 2 {
+		productName = fmt.Sprintf("Azure CDN from %s", s[1])
+		skuName = s[0]
+		if strings.ToLower(s[1]) == "verizon" {
+			name = fmt.Sprintf("Outbound data transfer (%s, ", s[0]+" "+s[1])
+		} else {
+			name = fmt.Sprintf("Outbound data transfer (%s, ", s[1])
+		}
+	}
+
+	data := []dataTier{
+		{name: fmt.Sprintf("%s%s", name, "first 10TB)"), startUsage: "0"},
+		{name: fmt.Sprintf("%s%s", name, "next 40TB)"), startUsage: "10000"},
+		{name: fmt.Sprintf("%s%s", name, "next 100TB)"), startUsage: "50000"},
+		{name: fmt.Sprintf("%s%s", name, "next 350TB)"), startUsage: "150000"},
+		{name: fmt.Sprintf("%s%s", name, "next 500TB)"), startUsage: "500000"},
+		{name: fmt.Sprintf("%s%s", name, "next 4000TB)"), startUsage: "1000000"},
+		{name: fmt.Sprintf("%s%s", name, "over 5000TB)"), startUsage: "5000000"},
+	}
+
+	meterName = fmt.Sprintf("%s Data Transfer", skuName)
+
+	var monthlyOutboundGb *decimal.Decimal
+	if u != nil && u.Get("monthly_outbound_gb").Type != gjson.Null {
+		monthlyOutboundGb = decimalPtr(decimal.NewFromInt(u.Get("monthly_outbound_gb").Int()))
+		tierLimits := []int{10000, 40000, 100000, 350000, 500000, 4000000}
+		tiers := usage.CalculateTierBuckets(*monthlyOutboundGb, tierLimits)
+
+		for i, d := range data {
```

Comment:
This is a nice way to handle so many tiers.  For some reason it never occurred to me to make a loop🙂